Only get data products with labels beginning with 'hlt'

Dr15Jones opened the pull request:
When run in fastsim jobs, TriggerSummaryProducerAOD was requesting
data products for non HLT modules. Now it only accepts data products
from modules with labels beginning with 'hlt'. In addition, a new
parameter was added to the module to allow explicitly specifying
modules which should be avoided when choosing data products.

Dr15Jones committed:
Only get data products with labels beginning with 'hlt'

When run in fastsim jobs, TriggerSummaryProducerAOD was requesting
data products for non HLT modules. Now it only accepts data products
from modules with labels beginning with 'hlt'. In addition, a new
parameter was added to the module to allow explicitly specifying
modules which should be avoided when choosing data products.

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for CMSSW_8_1_X.

It involves the following packages:

HLTrigger/HLTcore


@perrotta, @cmsbuild, @silviodonato, @Martin-Grunewald, @fwyzard, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ac03d9318cc8109a99c82818b1b55e4fbfac36e7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15463/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3cd4d481d5d8ec480351fc60f297b6aa44a07d72
https://github.com/cms-sw/cmssw/commit/ca4da2eec8ba279601a00973d174b5dca3864c9c
https://github.com/cms-sw/cmssw/commit/9c503f8e959fe9b190f0e5268b256bed8872d5ab
https://github.com/cms-sw/cmssw/commit/a08c942d5ca8149e144f657c076ac2d1b2d228d7
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15463/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15463/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15463/summary.html

Dr15Jones committed:
Use glob patterns when choosing module labels in TriggerSummaryProducerAOD

Use glob patterns to specify patterns to match and patterns to reject
for module labels when getting data products.

cmsbuild commented:
Pull request #16046 was updated. @perrotta, @cmsbuild, @silviodonato, @Martin-Grunewald, @fwyzard, @davidlange6 can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Dr15Jones commented:
I ran this code on workflow 135.1 and dumped the consumes information for TriggerSummaryProducerAOD. It was a very long list and all module labels began with 'hlt'.

cmsbuild commented:
+1
Tested at: 1a1279539840f8fadd61be01408780628e7e091f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15473/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16046/15473/summary.html

Dr15Jones commented:
ping
This is needed to further work on the next phase of the threaded framework (i.e. to fix most of the problems seen in CMSSW_8_1_DEVEL_X).

Martin-Grunewald commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

Dr15Jones commented:
@davidlange6 It would help to get this pull request merged sooner rather than later in order to further testing of the changes to the threaded framework in _DEVEL_X.

davidlange6 commented:
+1
